Lib/code.py

```diff
@@ -119,32 +118,35 @@ def showsyntaxerror(self, filename=None, **kwargs):
                 else:
                     # Stuff in the right filename
                     value = SyntaxError(msg, (filename, lineno, offset, line))
-            self._showtraceback(typ, value, None, colorize)
+            self._showtraceback(typ, value, None)
         finally:
             typ = value = tb = None
 
-    def showtraceback(self, **kwargs):
+    def showtraceback(self):
         """Display the exception that just occurred.
 
         We remove the first stack item because it is our own code.
 
         The output is written by self.write(), below.
 
         """
-        colorize = kwargs.pop('colorize', False)
         try:
             typ, value, tb = sys.exc_info()
-            self._showtraceback(typ, value, tb.tb_next, colorize)
+            self._showtraceback(typ, value, tb.tb_next)
         finally:
             typ = value = tb = None
 
-    def _showtraceback(self, typ, value, tb, colorize):
+    def _showtraceback(self, typ, value, tb, colorize=False, limit=None):
+        # This method is being overwritten in
+        # _pyrepl.console.InteractiveColoredConsole to pass different values of
+        # colorize and limit
         sys.last_type = typ
         sys.last_traceback = tb
         sys.last_exc = sys.last_value = value = value.with_traceback(tb)
         if sys.excepthook is sys.__excepthook__:
             lines = traceback.format_exception(typ, value, tb,
-                                                colorize=colorize)
+                                                colorize=colorize,
+                                                limit=limit)
             self.write(''.join(lines))
```

Comment:
I think that only this part of code should be overridden. I propose to extract these lines as `_excepthook()` and override in `InteractiveColoredConsole` without calling the superclass method, but with duplicating its code and adding new arguments for the `format_exception()` call.

This may be not perfect design, but it looks less ugly.

Reply:
I went with the suggestion and added an `_excepthook` method, looks much nicer.